### DISCUSSION TITLE
How to calculate diversity using abundance data with floats (percentage) instead of counts (integers)

### DISCUSSION THREAD

**Discussion by LenHC** [scikit-bio/scikit-bio#2041] [Q&A]
I am trying to calculate diversity using abundance data with floats (percentage) instead of counts (integers). For alpha diversity, the function says that the `counts` argument can be floats & abundance, but it raises an error trying to convert my abundance data of floats into integers. What would be a proper way to use the abundance data? I do not have the total counts data so I cannot calculate the counts from the abundance data.

**Comment by qiyunzhu** [comment]
Hi @LenHC I think most alpha diversity metrics in scikit-bio do accept floats. Some specific ones may not. For example:

```python
>>> import numpy as np
>>> from skbio.diversity.alpha import *
>>> comm = np.array([1.0, 1.5, 2.0, 3.0, 4.5])
>>> shannon(comm)
np.float64(1.480016885223025)
>>> simpson_d(comm)
np.float64(0.2534722222222222)
>>> pielou_e(comm)
np.float64(0.9195861945271688)
```

However, by definition, alpha diversity should start from abundance (count) data. It is may not appropriate to calculate alpha diversity based on percentages, which will be rounded to zero by some metrics, particularly richness metrics. But diversity and evenness metrics may work, as they naturally consider relative abundances of taxa. For example:

```python
>>> comm = np.array([0.1, 0.2, 0.3, 0.4])
>>> shannon(comm)
np.float64(1.2798542258336676)
>>> simpson(comm)
np.float64(0.7)
```

If you do intend to calculate richness metrics based on percentage data, I would recommend bringing up total abundance to one million (i.e., counts per million, or CPM) before calculation. Please note that the results are not real alpha diversity of the community, but they are cross-comparable among samples.

**Comment by qiyunzhu** [comment]
We hope the new [alpha diversity documentation page](https://scikit.bio/docs/latest/generated/skbio.diversity.alpha.html) of the just released scikit-bio 0.6.1 will be informative to you.
